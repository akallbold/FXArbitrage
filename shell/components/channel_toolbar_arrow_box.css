@import 'playa-styleguide/src/stylesheets/variables.css';

.wrapper {
  display: flex;
  justify-content: flex-start;
  align-content: center;
}

.box {
  position: relative;
  width: 55px;
  height: 50px;
  border: 1px solid;
}

.title {
  font-family: var(--font-family-regular);
  font-size: 14px;
  font-weight: 600;
  align-self: center;
  margin-left: 25px;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.box:after, .box:before {
  left: 53px;
  top: -1px;
  border-style: solid;
  border-color: transparent transparent transparent #007bff;
  border-width: 25px 0 25px 14px;
  content: " ";
  height: 0;
  width: 0;
  position: absolute;
  pointer-events: none;
}

.box:before {
  left: 54px;
  top: -1px;
}

.greyIcon {
  align-self: center;
  margin-right: 5px;
  width: 12px;
  height: 12px;
}

.greyIcon svg {
  width: 100%;
  height: 100%;
  fill: var(--text-grey-light);
}

.greyIcon .snapchatIcon {
  width: 15px;
  height:15px;
  margin-top: -1px;
}

.otherActiveChannels {
  display: flex;
  margin-left: 25px;
}

.snapchat {
  background: var(--snapchat-yellow);
  border-color: var(--snapchat-yellow-dark);
}
.snapchat:after {
  border-left-color: var(--snapchat-yellow);
}
.snapchat:before {
  border-left-color: var(--snapchat-yellow-dark);
}

.facebook {
  background: var(--facebook-blue);
  border-color: var(--facebook-blue-dark);
}
.facebook:after {
  border-left-color: var(--facebook-blue);
}
.facebook:before {
  border-left-color: var(--facebook-blue-dark);
}

.twitter {
  background: var(--twitter-blue);
  border-color: var(--twitter-blue-dark);
}
.twitter:after {
  border-left-color: var(--twitter-blue);
}
.twitter:before {
  border-left-color: var(--twitter-blue-dark);
}

.pinterest {
  background: var(--pinterest-red);
  border-color: var(--pinterest-red-dark);
}
.pinterest:after {
  border-left-color: var(--pinterest-red);
}
.pinterest:before {
  border-left-color: var(--pinterest-red-dark);
}

.colorIcon svg {
  width: 18px;
  height: 18px;
  top: 25px;
  left: 30px;
  position: absolute;
  margin-top: -9px;
  margin-left: -9px;
  fill: #fff;
}

.colorIcon .snapchatIcon {
  stroke: #000;
}

.colorIcon .instagramIcon {
  width: 11px;
  height: 11px;
  margin-left: 8px;
  margin-top: -6px;
}

.colorIcon .facebookIcon {
  width: 16px;
  height: 16px;
  margin-left: -15px;
  margin-top: -9px;
}

.colorIcon .twitterIcon {
}

.colorIcon .pinterestIcon {
}
