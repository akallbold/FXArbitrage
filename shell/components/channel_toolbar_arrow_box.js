import React from 'react'
import { PropTypes } from 'prop-types'
import { connect } from 'react-redux'
import {
  TwitterIcon,
  FacebookIcon,
  SnapchatIcon,
  PinterestIcon,
  InstagramIcon
} from 'playa-styleguide'
import { brandChannelsArray, currentPageTitleSelector } from 'selectors/org_selectors'
import { curChannelSelector } from 'selectors/platform_selectors'
import Styles from './channel_toolbar_arrow_box.css'
import { SNAPCHAT, TWITTER, FACEBOOK, PINTEREST, INSTAGRAM } from './shared_constants'

const channelIcons = {
  [SNAPCHAT]: <SnapchatIcon className={Styles.snapchatIcon} viewBox='0 0 18 17' preserveAspectRatio='xMidYMid' />,
  [TWITTER]: <TwitterIcon className={Styles.twitterIcon} />,
  [FACEBOOK]: <FacebookIcon className={Styles.facebookIcon} viewBox='0 0 8 15' preserveAspectRatio='xMidYMid' />,
  [PINTEREST]: <PinterestIcon className={Styles.pinterestIcon} />,
  [INSTAGRAM]: <InstagramIcon className={Styles.instagramIcon} viewBox='0 0 15 15' preserveAspectRatio='xMidYMid' />
}

export const ArrowBox = ({ channel, activeChannels, title }) => (
  <div className={Styles.wrapper}>
    <div className={`${Styles[channel]} ${Styles.box} ${Styles.colorIcon}`}>
      {channelIcons[channel] || null}
      {channel === FACEBOOK && channelIcons[INSTAGRAM]}
    </div>
    <div className={Styles.otherActiveChannels}>
      {
        activeChannels.map((channelName) => {
          if (
            channelName !== channel && // Do not show icon on it's own channel.
            !(channelName === INSTAGRAM && channel === FACEBOOK) // Don't show snapchat on facebook
          ) {
            return (
              <div key={channelName} className={`${Styles.greyIcon}`}>
                {channelIcons[channelName] || null}
              </div>
            )
          }
          return null
        })
      }
    </div>

    {title && <h1 className={Styles.title}>{title}</h1>}
  </div>
)

ArrowBox.defaultProps = {
  channel: 'facebook'
}

const { string, arrayOf } = PropTypes

ArrowBox.propTypes = {
  channel: string,
  activeChannels: arrayOf(string).isRequired
}

const mapStateToProps = state => ({
  channel: curChannelSelector(state),
  activeChannels: brandChannelsArray(state),
  title: currentPageTitleSelector(state)
})

export default connect(mapStateToProps)(ArrowBox)
