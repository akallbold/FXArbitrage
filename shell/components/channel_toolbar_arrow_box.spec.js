import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ArrowBox } from './channel_toolbar_arrow_box'

import {
  TWITTER,
  FACEBOOK,
  PINTEREST
} from './shared_constants'

describe('ArrowBox', () => {
  const activeChannels = [TWITTER, FACEBOOK, PINTEREST]

  it('renders correctly when snapchat', () => {
    const rapper = shallow(<ArrowBox channel='snapchat' activeChannels={activeChannels} />)
    expect(toJson(rapper)).toMatchSnapshot()
  })

  it('renders correctly when facebook', () => {
    const rapper = shallow(<ArrowBox channel='facebook' activeChannels={activeChannels} />)
    expect(toJson(rapper)).toMatchSnapshot()
  })

  it('renders correctly when twitter', () => {
    const rapper = shallow(<ArrowBox channel='twitter' activeChannels={activeChannels} />)
    expect(toJson(rapper)).toMatchSnapshot()
  })

  it('renders correctly when pinterest', () => {
    const rapper = shallow(<ArrowBox channel='pinterest' activeChannels={activeChannels} />)
    expect(toJson(rapper)).toMatchSnapshot()
  })

  it('renders with a title if there is a title', () => {
    const singer = shallow(<ArrowBox channel='snapchat' activeChannels={activeChannels} title='title' />)
    expect(toJson(singer)).toMatchSnapshot()
    expect(singer.find('.title').exists()).toBe(true)
  })
})
