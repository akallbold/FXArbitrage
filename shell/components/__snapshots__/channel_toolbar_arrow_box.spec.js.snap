// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowBox renders correctly when facebook 1`] = `
<div
  className="wrapper"
>
  <div
    className="facebook box colorIcon"
  >
    <test-file-stub
      className="facebookIcon"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 8 15"
    />
    <test-file-stub
      className="instagramIcon"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 15 15"
    />
  </div>
  <div
    className="otherActiveChannels"
  >
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="twitterIcon"
      />
    </div>
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="pinterestIcon"
      />
    </div>
  </div>
</div>
`;

exports[`ArrowBox renders correctly when pinterest 1`] = `
<div
  className="wrapper"
>
  <div
    className="pinterest box colorIcon"
  >
    <test-file-stub
      className="pinterestIcon"
    />
  </div>
  <div
    className="otherActiveChannels"
  >
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="twitterIcon"
      />
    </div>
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="facebookIcon"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 8 15"
      />
    </div>
  </div>
</div>
`;

exports[`ArrowBox renders correctly when snapchat 1`] = `
<div
  className="wrapper"
>
  <div
    className="snapchat box colorIcon"
  >
    <test-file-stub
      className="snapchatIcon"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 18 17"
    />
  </div>
  <div
    className="otherActiveChannels"
  >
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="twitterIcon"
      />
    </div>
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="facebookIcon"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 8 15"
      />
    </div>
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="pinterestIcon"
      />
    </div>
  </div>
</div>
`;

exports[`ArrowBox renders correctly when twitter 1`] = `
<div
  className="wrapper"
>
  <div
    className="twitter box colorIcon"
  >
    <test-file-stub
      className="twitterIcon"
    />
  </div>
  <div
    className="otherActiveChannels"
  >
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="facebookIcon"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 8 15"
      />
    </div>
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="pinterestIcon"
      />
    </div>
  </div>
</div>
`;

exports[`ArrowBox renders with a title if there is a title 1`] = `
<div
  className="wrapper"
>
  <div
    className="snapchat box colorIcon"
  >
    <test-file-stub
      className="snapchatIcon"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 18 17"
    />
  </div>
  <div
    className="otherActiveChannels"
  >
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="twitterIcon"
      />
    </div>
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="facebookIcon"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 8 15"
      />
    </div>
    <div
      className="greyIcon"
    >
      <test-file-stub
        className="pinterestIcon"
      />
    </div>
  </div>
  <h1
    className="title"
  >
    title
  </h1>
</div>
`;
